qemu/tests/seabios_strict.py

```diff
@@ -0,0 +1,101 @@
+import os
+import re
+import logging
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import data_dir
+from virttest import env_process
+
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    KVM seabios test:
+    1) Start guest with sga bios
+    2) Check the boot result when '-boot strict=on/off'
+       on:  Hard Disk -> NIC
+       off: Hard Disk -> NIC -> DVD/CD ...
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def create_cdrom():
+        """
+        Create 'test' cdrom
+        """
+        logging.info("creating test cdrom")
+        cdrom_test = params.get("cdrom_test", '/tmp/test.iso')
+        cdrom_test = utils_misc.get_path(data_dir.get_data_dir(), cdrom_test)
+        process.run("dd if=/dev/urandom of=test bs=10M count=1")
+        process.run("mkisofs -o %s test" % cdrom_test)
+        process.run("rm -f test")
+
+    def cleanup_cdrom():
+        """
+        Remove 'test' cdrom
+        """
+        logging.info("cleaning up test cdrom")
+        cdrom_test = utils_misc.get_path(data_dir.get_data_dir(),
+                                         params.get("cdrom_test"))
+        os.remove(cdrom_test)
+
+    def get_output(session_obj):
+        """
+        Use the function to short the lines in the scripts
+        """
+        return session_obj.get_stripped_output()
+
+    def boot_check():
+        """
+        boot_fail_infos check
+        """
+        return re.search(fail_infos, get_output(vm.serial_console), re.S)
+
+    def boot_check_extra():
+        """
+        boot_fail_infos_extra check
+        """
+        return re.search(fail_infos_extra, get_output(vm.serial_console), re.S)
```

Comment:
`boot_check()` and `boot_check_extra()` can be merged, and even can be dropped, you can use `lamda` in utils_misc.wait_for().

Reply:
@ybduan I don't understand why it is complex, they are 1. only one sentence and 2. totally same just with a variant change. Just simply replace the function name in your wait_for() with the actually executed line, with an extra `lambda: ` in front of it. Or at least it should be one single function with a parameter like `file_info`.
Please do avoid those one sentence functions in your code in the future, as well as redundant implementations. Thanks.

Reply:
@ybduan For 1, maybe, but not this one. And not most ones. Anyway I don't accept that without sufficient reason.
2 Acceptable to me. But for 3, the existence of _foo() does not make sense at all. Thanks.

Reply:
@ybduan By the way, I once got other maintainers complain about me accepting one of your previous pull request which has similar one sentence implementation, and I absolutely agree with that complain. So let's stop argue this please. Thank you.

Reply:
@ybduan I understand that directly use `re` in wait_for() will make lines too long which might annoy you, so I would accept the third solution if you remove the unreasonable `_foo()`, so directly
```
def check(info):
    return re.search(info, vm.serial_console.get_stripped_output(), re.S)
```
Thanks.

Reply:
If we remove it as you wish:

17:34:53 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-65.0-py2.7.egg/avocado_vt/test.py:435
17:34:53 ERROR| Traceback (most recent call last):                              
17:34:53 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-65.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
17:34:53 ERROR|     return fn(*args, **kwargs)                                  
17:34:53 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/seabios_strict.py", line 92, in run
17:34:53 ERROR|     if not utils_misc.wait_for(check(fail_infos), timeout, 1):  
17:34:53 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-65.0-py2.7.egg/virttest/utils_misc.py", line 685, in wait_for
17:34:53 ERROR|     output = func()                                             
17:34:53 ERROR| TypeError: 'NoneType' object is not callable

Reply:
@ybduan You need a `lamda:` in front of `check(fail_infos)`

Reply:
Updated